pkg/client/storage.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+type Storage interface {
```

Comment:
Looks iffy. `DSNer`?

Reply:
I feel this interface will grow in the future, ill keep it a storage for now.

Reply:
Why not have multiple interfaces providing multiple functionalities have its own interface (interfaces to be developed/identified as needed while we refactor) so that we know what it requires?

Reply:
Might as well just pass a string for now then no? i.e `dsn string` as this interface is not really useful.

Reply:
Yes and no, a `db.DB` or whatever can satisfy a number of `*Storage` interfaces but it cannot satisfy a `dsn string` input param.